docs/finality-provider-phase2.md

```diff
@@ -0,0 +1,589 @@
+--daemon-address 127.0.0.1:12581 \ 
+--chain-id bbn-test-5 \ 
+--commission 0.05 \ 
+--key-name finality-provider \ 
+--moniker "MyFinalityProvider" \ 
+--website "https://myfinalityprovider.com" \
+--security-contact "security@myfinalityprovider.com" \ 
+--details "finality provider for the Babylon network" \ 
+--home ./fp \ --passphrase "passphrase" 
+```
+
+Required parameters: 
+- `--chain-id`: The Babylon chain ID (`bbn-test-5`) 
+- `--commission`: The commission rate (between 0 and 1) that you'll receive from
+delegators 
+- `--key-name`: Name of the key in your keyring for signing
+transactions - 
+`--moniker`: A human-readable name for your finality provider
+
+Optional parameters: 
+- `--website`: Your finality provider's website 
+- `--security-contact`: Contact email for security issues 
+- `--details`:
+Additional description of your finality provider 
+- `--daemon-address`: RPC
+address of the finality provider daemon (default: 127.0.0.1:12581)
+
+Upon successful creation, the command will return a JSON response containing
+your finality provider's details:
+
+``` json
+{
+    "fp_addr": "bbn1ht2nxa6hlyl89m8xpdde9xsj40n0sxd2f9shsq", 
+    "btc_pk_hex":
+    "cf0f03b9ee2d4a0f27240e2d8b8c8ef609e24358b2eb3cfd89ae4e4f472e1a41",
+      "description": 
+      {
+        "moniker": "MyFinalityProvider", 
+        "website": "https://myfinalityprovider.com", 
+        "security_contact": "security@myfinalityprovider.com", 
+        "details": "finality provider for the Babylon network"
+      }, 
+      "commission": "0.050000000000000000", 
+      "status": "CREATED"
+} 
+```
+
+The response includes: 
+- `fp_addr`: Your Babylon account address for receiving
+rewards 
+- `btc_pk_hex`: Your unique BTC public key identifier (needed for
+registration) 
+- `description`: Your finality provider's metadata 
+- `commission`:
+Your set commission rate 
+- `status`: Current status of the finality provider
+
+## Register Finality Provider
+
+The `register-finality-provider` command registers your finality provider on the
+Babylon chain. This command requires:
+
+1. The BTC public key (obtained from the `create-finality-provider` command) 
+2. A funded Babylon account (needs BBN tokens for transaction fees) 
+3. A running FPD daemon
+
+``` shell
+fpd register-finality-provider \
+cf0f03b9ee2d4a0f27240e2d8b8c8ef609e24358b2eb3cfd89ae4e4f472e1a41 \
+--daemon-address 127.0.0.1:12581 \ 
+--passphrase <passphrase> \ 
+--home <path> \ 
+```
+
+> Note: The BTC public key (`cf0f03...1a41`) is obtained from the previous
+`create-finality-provider` command.
+
+If successful, the command will return a transaction hash:
+
+``` shell
+{ "tx_hash":
+"C08377CF289DF0DC5FA462E6409ADCB65A3492C22A112C58EA449F4DC544A3B1" } 
+```
+
+ You can query this hash to confirm the transaction was successful by navigating
+ to the babylon chain and making a query, such as below:
+
+```shell 
+babylond query tx <transaction-hash> --chain-id bbn-test-5 
```

Comment:
I don't think we should paste `babylond` commands in this guide. We can just say that this is a transaction hash and you can verify its success by looking it up.